apps/predbat/predbat.py

```diff
@@ -8900,9 +8909,30 @@ def fetch_config_options(self):
         self.set_inverter_notify = self.get_arg("set_inverter_notify")
         self.set_discharge_freeze_only = self.get_arg("set_discharge_freeze_only")
         self.set_discharge_during_charge = self.get_arg("set_discharge_during_charge")
-        self.set_charge_window = self.get_arg("set_charge_window")
-        self.calculate_best_charge = self.get_arg("calculate_best_charge")
-        self.calculate_best_discharge = self.get_arg("calculate_best_discharge")
+
+        # Mode
+        self.predbat_mode = self.get_arg("mode")
+        if self.predbat_mode == PREDBAT_MODE_OPTIONS[PREDBAT_MODE_CONTROL_SOC]:
+            self.calculate_best_charge = True
+            self.calculate_best_discharge = False
+            self.set_charge_window = False
+        elif self.predbat_mode == PREDBAT_MODE_OPTIONS[PREDBAT_MODE_CONTROL_CHARGE]:
+            self.calculate_best_charge = True
+            self.calculate_best_discharge = False
+            self.set_charge_window = True
+        elif self.predbat_mode == PREDBAT_MODE_OPTIONS[PREDBAT_MODE_CONTROL_CHARGEDISCHARGE]:
+            self.calculate_best_charge = True
+            self.calculate_best_discharge = True
+            self.set_charge_window = True
+        else:  # PREDBAT_MODE_OPTIONS[PREDBAT_MODE_MONITOR]
```

Comment:
Personally I'd make this an explicit `elif`, followed by an `else` that throws an error due to an unexpected mode, so it's safer in the future if you add other values to the mode.

Or is this `else` primarily the mechanism to set the mode to MONITOR if no mode is currently set?